dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -661,6 +663,78 @@ def test_sample_size_passed_to_profile(self, *mocks):
         profiler._sampling_ratio = 0.2
         self.assertEqual(10000, update_mock.call_args[0][1])
 
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnPrimitiveTypeProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnStatsProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    def test_index_overlap_resolved_for_update_profile(self, *mocks):
+        data = pd.Series([0, None, 1, 2, None])
+        profile = StructuredDataProfile(data)
+        self.assertEqual(0, profile.min_id)
+        self.assertEqual(4, profile.max_id)
```

Comment:
should max_id be 3 (number 2) in this case?

Reply:
or you mean min_id as start_id and max_id as end_id?

Reply:
Yes it doesn't have to do with non-None or anything like that, its purely just to remember the range of the index.